foundationdb/src/api.rs

```diff
@@ -116,214 +342,107 @@ pub struct NetworkBuilder {
-    /// net_thread.join().expect("failed to join fdb thread");
-    /// ```
-    #[allow(clippy::mutex_atomic)]
-    pub fn build(self) -> FdbResult<(NetworkRunner, NetworkWait)> {
-        unsafe { error::eval(fdb_sys::fdb_setup_network())? }
-
-        let cond = Arc::new((Mutex::new(false), Condvar::new()));
-        Ok((NetworkRunner { cond: cond.clone() }, NetworkWait { cond }))
+    /// FoundationDB run loop. The call is now an noop, it has been kept for compatibility reason
+    pub fn build(self) -> FdbResult<()> {
+        Ok(())
     }
 
-    /// Starts the FoundationDB run loop in a dedicated thread.
-    /// This finish initializing the FoundationDB Client API and can only be called once per process.
-    ///
-    /// # Returns
-    ///
-    /// A `NetworkAutoStop` handle which must be dropped before the program exits.
-    ///
-    /// # Safety
-    ///
-    /// You *MUST* ensure `drop` is called on the returned object before the program exits.
-    /// This is not required if the program is aborted.
-    ///
-    /// This method used to be safe in version `0.4`. But because `drop` on the returned object
-    /// might not be called before the program exits, it was found unsafe.
-    ///
-    /// # Panics
-    ///
-    /// Panics if the dedicated thread cannot be spawned or the internal condition primitive is
-    /// poisonned.
-    ///
-    /// # Examples
+    /// Starts the FoundationDB network thread in a dedicated thread.
+    /// This finish initializing the FoundationDB Client API.
     ///
-    /// ```rust
-    /// use foundationdb::api::FdbApiBuilder;
+    /// You **must** call `stop_network` at the end of your program, or use the Drop feature of [NetworkAutoStop]
     ///
-    /// let network_builder = FdbApiBuilder::default().build().expect("fdb api initialized");
-    /// let network = unsafe { network_builder.boot() };
-    /// // do some interesting things with the API...
-    /// drop(network);
-    /// ```
+    /// ## Safety
     ///
-    /// ```rust
-    /// use foundationdb::api::FdbApiBuilder;
-    ///
-    /// #[tokio::main]
-    /// async fn main() {
-    ///     let network_builder = FdbApiBuilder::default().build().expect("fdb api initialized");
-    ///     let network = unsafe { network_builder.boot() };
-    ///     // do some interesting things with the API...
-    ///     drop(network);
-    /// }
-    /// ```
+    /// This function is `unsafe` because Rust cannot enforce
+    /// that `stop_network` will be called at the end of your program.
+    /// Using this function incorrectly may introduce undefined behavior.
     pub unsafe fn boot(self) -> FdbResult<NetworkAutoStop> {
-        let (runner, cond) = self.build()?;
-
-        let net_thread = runner.spawn();
-
-        let network = cond.wait();
-
-        Ok(NetworkAutoStop {
-            handle: Some(net_thread),
-            network: Some(network),
-        })
+        spawn_network_thread()?;
+        Ok(NetworkAutoStop::new())
```

Comment:
The logic is that a NetworkAutoStop should only be created IF the network is running ?
There is a race condition here, that can lead to a false sense of network thread running.

Thread 1:  auto g = boot()
Thread 2 : boot() -> spawn_network_thread()
Thread 1:  drop(g)
Thread 2 : boot() return NetworkAutoStop::new()
  
Thread 2: